src/query/storages/index/benches/build_from_block.rs

```diff
@@ -44,17 +44,19 @@ fn bench_u64(c: &mut Criterion) {
     let column = block.try_column_by_name("a").unwrap();
 
     let mut builder = Xor8Builder::create();
-    builder.add_keys(&column.to_values());
+    let values: Vec<DataValue> = (0..column.len()).map(|i| column.get(i)).collect();
+    builder.add_keys(&values);
     let filter = builder.build().unwrap();
 
-    for key in column.to_values() {
-        assert!(filter.contains(&key), "key {} present", key);
+    for key in &values {
+        assert!(filter.contains(key), "key {} present", key);
     }
 
     c.bench_function("xor8_filter_u64_1m_rows_build_from_column_to_values", |b| {
         b.iter(|| {
             let mut builder = Xor8Builder::create();
-            builder.add_keys(&criterion::black_box(column.to_values()));
+            let values: Vec<DataValue> = (0..column.len()).map(|i| column.get(i)).collect();
```

Comment:
This collect also do copy, it sames as to_values

Reply:
https://github.com/datafuselabs/databend/pull/9053/files#diff-4381823528cbebdb8c595a2cf1d9bcc694ecc5dcc872963dedf7b9bb4ef70e51R132